### DISCUSSION TITLE
q parameter for the same species

### DISCUSSION THREAD

**Discussion by Nyanaaa** [pyscal/pyscal#129] [Q&A] [answered]
Hello, 

I have a new question about your code.

When dealing with alloys, it can be useful to estimate the bond orientational order parameter (the q parameter) for atoms of the same species. 

To do so, we need to extract the neighbors list for the system, and then estimate the q parameters of an atom only by taking into account the atoms of the same species.

For instance, the atom i of type 1 has 10 neighbors, 6 of type 1 and 4 of type 2, then we can estimate its q parameters by considering only its neighbors of type 1. (for instance they use it on this paper on page 11 doi: 10.1063/1.5118867)

Is there a way to do it with pyscal while using an adaptative or sann cutoff? 

I tried but I could not manage to do it. 

Is it for instance possible to modify "manually" the neighbors list measure for the whole system in order to suppress the neighbors I am not interested in? 

I hope my question is clear.

Sincerely yours

**Comment by srmnitc** [comment, marked as answer]
@Nyanaaa Yes, you can do this. In the `find_neighbors` function, you can add the optional keyword `filter="type"`. This takes only atoms of the same type while finding neighbors. If you want only atoms of different species, you can use `"type_r"`. For more details, see [here](https://pyscal.org/en/latest/pyscal.html?highlight=find_neighbors#pyscal.core.System.find_neighbors).

**Reply by Nyanaaa** [reply]
Thanks, on my computer, it does not work with cutoff='sann', but it works with cutoff='adaptive' so I can use it, thanks :)
